sdk/nodejs/tests/cmd/run/error.spec.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as assert from "assert";
+import * as util from "util";
+import { defaultErrorMessage } from "../../../cmd/run/error";
+
+describe("defaultErrorMessage", () => {
+    function doThrow(thingToThrow: any): string {
+        try {
+            throw thingToThrow;
+        } catch (err) {
+            return defaultErrorMessage(err);
+        }
+    }
+
+    class CustomError extends Error {
+        constructor(message: string) {
+            super(`a custom error: ${message}`);
+        }
+    }
+
+    // This object has no toString, and forces an error when inspected.
+    // It is very naughty.
+    const veryNaughty = Object.assign(Object.create(null), {
+        [util.inspect.custom]() {
```

Comment:
Hah, impressive that one can even do that.

Reply:
Yeah, I was going to add a comment "inspect never crashes", and then I doubted that and did some research :D